packages/components/src/utils/event-guard.ts

```diff
@@ -0,0 +1,15 @@
+export function eventGuard(
+  event: CustomEvent,
+  callback: () => void,
+  options: { targetLocalName?: string; delegatorSelector?: string } = {}
```

Comment:
```suggestion
  options: { targetLocalName!: string; delegatorSelector?: string } = {}
```
The guard does not make any sence without the targetLocalName being specified.
Since this is the case, the options object is always needed, why I would suggest to set it as the second parameter. Because it will be better readable, when you use the utility if you first specify the event, then the options object, then the callback, which will take multiple lines in most cases.